### PR TITLE
Tests & refactor

### PR DESCRIPTION
**Pull request by elenatorro**
- Unit tests.
- Using objects instead of an array to store the alphabets in order to access them faster.

Hi! I was looking for something like that to use it in a project, I made some irrelevant changes, but I also added tests

Regards,

Elena.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,83 +6,173 @@
...
-				letter = this.sets.ascii[i];
-			}
-		}
-		
-		// if not found will return undefined
-		return letter;
-	},
+module.exports = {
+  convert: function (character) {
+    return !!BRAILLE[character] ? BRAILLE[character] : " ";
```
Comment (Nonemoticoner) [inline index.js]: Double `not`? I'd recommend `Object.prototype.hasOwnProperty()` instead.

**Comment by Nonemoticoner** [comment]
Hi,

I'm glad for your contribution. 

There's refactor required though. You forgot to exclude the `node_modules` directory from your commit. Also, add packages to `package.json`/`devDependencies` those were used for tests.

I commented on that weird double not (`!!`).

I'm not sure if object is going to be faster than string which is a primitive but I'm willing to have some benchmarking very soon to investigate it.

File: index.js (hunk 1 above)
Comment (elenatorro) [inline index.js]: Ok, we have to think first what to return if there is a character without equivalence, question mark, blank space or something else. What I was doing was to check if the property exists and that doesn't contain a falsy value (the double `!!` coerces the value to a boolean). If you want to test that a key exists, even if its value is falsy, then is ok to use `Object.prototype.hasOwnProperty.call(BRAILLE, i)`.  Hope it helped :)

**Comment by elenatorro** [comment]
I'll be glad to see the results, I haven't tested it and compared both scripts using a huge text. Sorry for missing the .gitignore file!

File: index.js (hunk 1 above)
Comment (Nonemoticoner) [inline index.js]: I think `?` will do. There seems to be no question mark in Braille alphabet as well as it is a common practice to use a question mark as replacement for unsupported signs in many applications e.g. MySQL, CLI.

JS will do the auto coertion anyway. No need for double not imo. So I'd stick to what I have previously written i.e. `return BRAILLE[character] || "?";`.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+node_modules
```
Comment (Nonemoticoner) [inline .gitignore]: The PR still has `node_modules` directory commited. You have to remove it :)

**Pull request merged**